sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -154,24 +174,175 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
             .andWhere("offerings.institutionLocation.id = :locationId", {
               locationId,
             }),
         "totalofferings",
       )
-      .where("programs.institution.id = :institutionId", { institutionId })
-      .getRawMany();
+      .where("programs.institution.id = :institutionId", { institutionId });
+
+    const queryParams: any[] = [...offeringTypes, locationId, institutionId];
+    // program name search
+    if (paginationOptions.searchName) {
+      summaryResult.andWhere("programs.name Ilike :searchProgramName", {
+        searchProgramName: `%${paginationOptions.searchName}%`,
+      });
+      queryParams.push(`%${paginationOptions.searchName}%`);
+    }
+
+    // total raw count
+    const totalCount = await getRawCount(
+      this.repo,
+      summaryResult.getSql(),
+      queryParams,
+    );
+
+    // pagination
+    summaryResult
+      .take(paginationOptions.pageLimit)
+      .skip(paginationOptions.page * paginationOptions.pageLimit);
+
+    // sort
+    if (paginationOptions.sortField && paginationOptions.sortOrder) {
+      summaryResult.orderBy(
+        databaseFieldOfInstitutionProgramDataTable(paginationOptions.sortField),
+        paginationOptions.sortOrder,
+      );
+    } else {
+      // default sort and order
+      summaryResult.orderBy(
+        databaseFieldOfInstitutionProgramDataTable(DEFAULT_SORT_FIELD),
+        FieldSortOrder.ASC,
+      );
+    }
+
+    const programs = await summaryResult.getRawMany();
 
-    return summaryResult.map((summary) => {
+    const programSummary = programs.map((summary) => {
       const summaryItem = new EducationProgramsSummary();
       summaryItem.id = summary.id;
       summaryItem.name = summary.name;
       summaryItem.cipCode = summary.cipcode;
       summaryItem.credentialType = summary.credentialtype;
+      summaryItem.credentialTypeToDisplay = credentialTypeToDisplay(
+        summary.credentialtype,
+      );
       summaryItem.approvalStatus = summary.approvalstatus;
       summaryItem.totalOfferings = summary.totalofferings;
       return summaryItem;
     });
+    return {
+      programsSummary: programSummary,
+      totalProgram: totalCount[0].count,
+    };
+  }
+  /**
+   * Gets all the programs that are associated with an institution
+   * alongside with the total of offerings on a particular location.
+   * @param institutionId Id of the institution.
+   * @param offeringTypes OfferingTypes array.
+   * @param paginationOptions pagination options
+   * @returns summary for location
+   */
+  async getPaginatedProgramsForAEST(
+    institutionId: number,
+    offeringTypes: OfferingTypes[],
+    paginationOptions: PaginationOptions,
+  ): Promise<ProgramsSummaryPaginated> {
+    // default data table sort field
+    const sortByColumn = "programs.createdAt"; //Default sort column
+    const paginatedProgramQuery = this.repo
+      .createQueryBuilder("programs")
+      .select([
+        "programs.id as programId",
```

Comment:
If the select/addSelect is used here (e.g. addSelect("programs.name", "programName")) it could avoid using the all lower case properties at line 328 on. Here there will be maybe a confusion in the future on why the alias is declared with the correct case but the properties have all lower case letters.